atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -194,4 +223,58 @@ private Column makeColumn(long ts) {
     private static ByteBuffer getRowName() {
         return ByteBuffer.wrap(PtBytes.toBytes(ROW_AND_COLUMN_NAME));
     }
+
+    private String replaceBracesWithStringFormatSpecifier(String msg) {
+        return msg.replaceAll("\\{\\}", "%s");
+    }
+
+    private String getCurrentTimestampValues(CASResult result) {
+        return result.current_values.stream()
+                .map(Column::getValue)
+                .map(PtBytes::toString)
+                .collect(Collectors.joining(", "));
+    }
+
+    @VisibleForTesting
+    long getId() {
+        return id;
+    }
+
+    private static final class IdAndTimestamp {
```

Comment:
This class is very similar to the format I used for keeping lock ID and reason together in the backup lock, although the types of each half are different. We should consider if we can extract this class (to a public `@Value.Immutable` class) and use it in both places.

We might have to parameterise it, and therefore use a `Long` instead of a `long` for the timestamp. But having`Long` and `long` in the same two-variable class seems a bit weird anyway.